package/MDAnalysis/core/groups.py

```diff
@@ -1923,6 +2000,149 @@ def __getattr__(self, attr):
+    # latter inherited from AtomGroup). Because of this bugs introduced in the
+    # class that cause an AttributeError may be very hard to diagnose and
+    # debug: the most obvious symptom is an infinite loop going through both
+    # __getattribute__ and __getattr__, and a solution might be to add said
+    # attribute to _UAG_SHORTCUT_ATTRS.
+
+    def __init__(self, base_group, selections, strings):
+        """
+
+        Parameters
+        ----------
+        base_group : :class:`AtomGroup`
+            group on which *selections* are to be applied.
+        selections : a tuple of :class:`~MDAnalysis.core.selection.Selection` instances
+            selections ready to be applied to *base_group*.
+
+        """
+        # Because we're implementing __getattribute__, which needs _u for
+        # its check, no self.attribute access can be made before this line
+        self._u = base_group.universe
+        self._selections = selections
+        self.selection_strings = strings
+        self._base_group = base_group
+        self._lastupdate = None
+        self._derived_class = base_group._derived_class
+        if self._selections:
+            # Allows the creation of a cheap placeholder UpdatingAtomGroup
+            # by passing an empty selection tuple.
+            self.ensure_updated()
+
+    def update_selection(self):
+        """
+        Forces the reevaluation and application of the group's selection(s).
+
+        This method is triggered automatically when accessing attributes, if
+        the last update occurred under a different trajectory frame.
+
+        """
+        bg = self._base_group
+        sels = self._selections
+        if sels:
+            # As with select_atoms, we select the first sel and then sum to it.
+            ix = sum([sel.apply(bg) for sel in sels[1:]],
+                     sels[0].apply(bg)).ix
+        else:
+            ix = np.array([], dtype=np.int)
+        # Run back through AtomGroup init with this information to remake ourselves
+        super(UpdatingAtomGroup, self).__init__(ix, self._u)
+        self.is_uptodate = True
+
+    @property
+    def is_uptodate(self):
+        """
+        Checks whether the selection needs updating
+
+        Returns
+        -------
+        bool
+            `True` if the group's selection is up-to-date, `False` otherwise.
+
+        """
+        try:
+            return self._u.trajectory.frame == self._lastupdate
+        except AttributeError: # self._u has no trajectory
+            return self._lastupdate == -1
+
+    @is_uptodate.setter
+    def is_uptodate(self, value):
+        if value:
+            try:
+                self._lastupdate = self._u.trajectory.frame
+            except AttributeError: # self._u has no trajectory
+                self._lastupdate = -1
+        else:
+            # This always marks the selection as outdated
+            self._lastupdate = None
+
+    def ensure_updated(self):
```

Comment:
So my final (I promise) comment is that maybe this method should get demoted to `_ensure_updated` to avoid potential confusion.  I think the code block below is perfectly reasonable, but won't work as `ensure_updated` doesn't detect when the `ts` was modified, just frame numbers.

```python
uag = u.select_atoms('something', updating=True)

for ts in u.trajectory:
    # access the UAG here for something minor
    something = uag.names

    # manipulate the trajectory, maybe a fitting or PBC transformation..
    u.atoms.pack_into_box()

    # now I want to ensure my selection is updated, so I guess I'll call..
    uag.ensure_updated()

    do_heavy_analysis(uag)  # whoops, this isn't actually updated though
```

I'd rather just give a single method for this (`update_selection`) rather than two which sound like they do the same thing, but are subtlely different.